docs/ruleset_functions/baseline_systems/baseline_hvac_sub_functions/are_all_terminal_types_CAV_With_None_Equal_to_Null.md

```diff
@@ -0,0 +1,24 @@
+# are_all_terminal_types_CAV_With_None_Equal_to_Null
+
+**Description:** Returns TRUE if all of the terminal unit types input to this function are constant air volume (CAV). It returns FALSE if any of the terminal units are of a type other than constant air volume (CAV).   
+
+**Inputs:**  
+- **B-RMR**: To evaluate if the terminal unit types are constant air volume (CAV).    
+- **terminal_unit_id_list**: List of terminal units to assess.  
+
+**Returns:**  
+- **aare_all_terminal_types_CAV_With_None_Equal_to_Null**: The function returns TRUE if all of the terminal unit types input to this function are constant air volume (CAV). It returns FALSE if any of the terminal units are of a type other than constant air volume (CAV).      
+ 
+**Function Call:**  None        
+
+## Logic:   
+- Set are_all_terminal_types_CAV_With_None_Equal_to_Null = TRUE: `are_all_terminal_types_CAV_With_None_Equal_to_Null = TRUE`  
+- For each terminal_b in the list of terminal units: `For terminal_b in terminal_unit_id_list:`  
+    - Create an object for the terminal unit (however the RCT accomplishes this): `terminal_b = terminal_b.id`  
+    - Check if the terminal unit type is not constant air volume (CAV): `if terminal_b.type != "CONSTANT_AIR_VOLUME":`  
+        - Set are_all_terminal_types_CAV_With_None_Equal_to_Null = FALSE: `are_all_terminal_types_CAV_With_None_Equal_to_Null = FALSE`
```

Comment:
@claperle the function looks good, not sure what the 'With_None_Equal_to_Null' means

Reply:
@supriyagoel This function is only for System 1a, 1c, System 10, and System 9b where the system is defined at the terminal so its my understanding that we don't want to allow the terminal.type to equal null **or** CV which we are allowing for all other systems defined that the HVAC system level. This 'With_None_Equal_to_Null' is to differentiate this function from the one that allows terminal.type to equal null or CV for a pass.

Reply:
As discussed during the RDS call, this function doesn't check fan control at HVAC system level, only at the terminal level where it checks both fan control and terminal type. The other function #513 'are_all_terminal_types_CAV.md ' checks for fan control at HVAC system level and terminal = CAV or NULL. both functions serve different purposes